client/store/modules/course/mutations.js

```diff
@@ -0,0 +1,37 @@
+import compact from 'lodash/compact';
+import transform from 'lodash/transform';
+import Vue from 'vue';
+
+export const upsertUser = (state, user) => {
+  Vue.set(state.users, user.id, user);
+};
```

Comment:
Also this can be one liner. He can change it according to this type of cases in other files. What do you think @vladimyr ?